index.bs

```diff
@@ -4225,24 +4230,38 @@ To <dfn export>run WebDriver BiDi preload scripts</dfn> given |environment setti
        |browsing context|. Otherwise let |realm| be |environment settings|'
        [=realm execution context=]'s Realm component.
 
-    1. Let |source| be |preload script|'s <code>expression</code>.
+    1. Let |base URL| be the [=API base URL=] of |environment settings|.
 
     1. Let |options| be the [=default classic script fetch options=].
 
-    1. Let |base URL| be the [=API base URL=] of |environment settings|.
+    1. Let |function declaration| be |preload script|'s <code>function declaration</code>.
 
-    1. Let |script| be the result of [=create a classic script=] with |source|,
+    1. Let (|script|, |function body evaluation status|) be the result of
+       [=evaluate function body=] with |function declaration|,
        |environment settings|, |base URL|, and |options|.
 
-    1. [=Prepare to run script=] with |environment settings|.
+    1. If |function body evaluation status| is an [=abrupt completion=], then
+       [=report the exception=] given by |function body evaluation status|.\[[Value]]
+       for |script|.
+
+    1. Let |function object| be |function body evaluation status|.\[[Value]].
+
+    1. If [=IsCallable=](|function object|) is <code>false</code>:
 
-    1. Let |evaluation status| be [=ScriptEvaluation=](|script|'s record).
+       1. Let |error| be a new [=TypeError=] object in |realm|.
 
-    1. If |evaluation status| is an [=abrupt completion=], then [=report the
-       exception=] given by |evaluation status|.\[[Value]] for |script|.
+       1. [=Report the exception=] |error|.
```

Comment:
Oh, I missed to ask if this error should actually be a dedicated WebDriver specific error or if `TypeError` is fine here which later would be converted to a `UnknownError` most likely.

Reply:
It doesn't ever get converted into a WebDriver error; there's no command in progress at the point at which preload scripts are run. Instead an exception is reported to the console and via `log.logEntryAdded` if the client is subscribed. We _could_ theoretically define some new JS error types for exceptions that happen in WebDriver scripts, but I don't think that's a good idea (UAs can already do things like replace the script filename by something that indicates a preload script in the stack trace).